persistence-api/src/main/java/io/stargate/db/datastore/DataStore.java

```diff
@@ -74,136 +57,45 @@ static DataStore create(Persistence.Connection connection, @Nonnull Parameters q
-   *
-   * <p>This is a shortcut for {@link #query(String, UnaryOperator, Object...)} where the data store
-   * default parameters are only modified to use the provided consistency level.
-   */
-  default CompletableFuture<ResultSet> query(
-      String queryString, ConsistencyLevel consistencyLevel, Object... values) {
-    return query(queryString, p -> p.withConsistencyLevel(consistencyLevel), values);
-  }
+  TypedValue.Codec valueCodec();
 
-  /**
-   * Executes the provided query against this data store.
-   *
-   * @param queryString the query to execute.
-   * @param parametersModifier a function called on the default parameters of this data store (the
-   *     instance provided when building the data store) and whose result parameters are used for
-   *     the query execution.
-   * @param values the (positional) values for the bind variables in {@code queryString}, if any.
-   * @return a future with a {@link ResultSet} object to access the result of the query. The future
-   *     is complete as soon as some initial result for the query is available, which for paging
-   *     queries means only the first page of the result set. As for {@link
-   *     Persistence.Connection#execute}, this future can be completed on a sensitive thread and one
-   *     should not chain blocking operations on this future <b>including</b> iterating over the
-   *     whole result set (as this may be block when paging kicks in to query further pages). In
-   *     other words, <b>do not</b> do:
-   *     <pre>
-   *   query(...).thenAccept(rs -> { for (Row r : rs) {...} });
-   * </pre>
-   *     Use {@link CompletableFuture#thenAcceptAsync} instead in that case.
-   */
-  CompletableFuture<ResultSet> query(
-      String queryString, UnaryOperator<Parameters> parametersModifier, Object... values);
-
-  /** Prepares the provided query against this data store. */
-  CompletableFuture<PreparedStatement> prepare(String queryString);
-
-  /**
-   * Executes the provided bound statements as a batch against this data store.
-   *
-   * <p>This is a shortcut for {@link #batch(List, UnaryOperator)} where the data store default
-   * parameters are only modified to use the provided consistency level.
-   */
-  default CompletableFuture<ResultSet> batch(
-      List<PreparedStatement.Bound> statements, ConsistencyLevel consistencyLevel) {
-    return batch(statements, p -> p.withConsistencyLevel(consistencyLevel));
-  }
-
-  /**
-   * Executes the provided bound statements as a batch against this data store.
-   *
-   * <p>This is a shortcut for {@link #batch(List, BatchType, UnaryOperator)} where batch type
-   * defaults to "logged".
-   */
-  default CompletableFuture<ResultSet> batch(
-      List<PreparedStatement.Bound> statements, UnaryOperator<Parameters> parametersModifier) {
-    return batch(statements, BatchType.LOGGED, parametersModifier);
+  /** Create a query using the DSL builder. */
+  default QueryBuilder queryBuilder() {
```

Comment:
I do not feel very strong about this, but the old method name of `query()` was leading to nicer flowable syntax, IMHO.

`dataStore.query().select()....execute()` (roughly) vs. `dataStore.queryBuilder().select()....build().execute()`... in the new code we have builder/build twice, plus the return type of `queryBuilder()` already identifies the objects as a "builder"

Reply:
My 2 cents: I find `queryBuilder()` more readable because it's more explicit. `query()` could mean "get the query" or "execute a query".
The builder/build repetition doesn't bother me, a lot of builder APIs work that way, for example the Guava immutable collections.

Reply:
Thanks @olim7t, you read my mind.

As a small anecdote, before this patch, `DataStore` has both the `query()` method, no args, that create a query builder, and a `query(String)` method that does actual querying :). That patch does rename the querying ones to `execute` (we were inconsistent before) so I guess `query()` for the query builder is a bit less confusing, but I still also prefer being a bit more explicit.